components/DependencyInjector/index.js

```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+export default class DependencyInjector extends React.Component {
```

Comment:
yeah this is a great proof of concept, but would be good to see an example we could use widely.